sdk/cosmosdb/cosmos/src/documents/PartitionKey.ts

```diff
@@ -1,5 +1,50 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { PartitionKeyDefinition } from "./PartitionKeyDefinition";
 
-export type PartitionKey = PartitionKeyDefinition | string | number | unknown;
+import { NonePartitionKeyLiteral, NullPartitionKeyLiteral } from "./PartitionKeyInternal";
+
+export type PartitionKey = PrimitivePartitionKeyValue | PrimitivePartitionKeyValue[];
+
+/**
+ * A primitive Partition Key value.
+ */
+export type PrimitivePartitionKeyValue =
+  | string
+  | number
+  | boolean
+  | NullPartitionType
+  | NonePartitionKey;
+
+/**
+ * The returned object represents a partition key value that allows creating and accessing items
+ * with a null value for the partition key.
+ */
+export type NullPartitionType = null;
```

Comment:
What's the value of having this alias instead of just using `null` in  `PrimitivePartitionKeyValue`?